### PR TITLE
feat: added breadcrumbs to bucket and data resource creation page

### PR DESCRIPTION
**Pull request by debjyoti-pandit**
The bucket creation page with breadcrumbs looks like the below - 

image

The data source creation page with bread crumbs looks like this - 

image

### CODE REVIEW

File: packages/components/bucket-policy/create-bucket-policy.tsx [hunk 1]
```diff
@@ -128,6 +129,21 @@ const CreateBucketPolicy: React.FC<CreateBucketPolicyProps> = ({
   const { ns = DATA_FEDERATION_NAMESPACE } = match.params;
 
   const { t } = useTranslation();
+  const breadcrumbs = [
+    {
+      name: DATA_FEDERATION,
```
Comment (GowthamShanmugam) [inline packages/components/bucket-policy/create-bucket-policy.tsx:134]: Data federation doesn't need translation?

File: packages/components/bucket-policy/create-bucket-policy.tsx [hunk 2]
```diff
@@ -168,20 +184,17 @@ const CreateBucketPolicy: React.FC<CreateBucketPolicyProps> = ({
...
-        </Title>
-        <p className="help-block">
+      <PageHeading
+        breadcrumbs={breadcrumbs}
+        title={t('Create new Bucket')}
+        className="breadcrumbs-header"
+      >
+        <p>
           {t(
             'A set of policies that would apply to all buckets (OBCs) created with the specific bucket policy. These policies include namespace and caching'
```
Comment (GowthamShanmugam) [inline packages/components/bucket-policy/create-bucket-policy.tsx:194]: Just a doubt:
  Will the user know what is OBC means? are we documenting this somewhere?

File: packages/components/data-resource/create-data-resource.tsx [hunk 3]
```diff
@@ -22,22 +23,34 @@ const CreateDataResource: React.FC<CreateDataResourceProps> = ({
   const { ns = DATA_FEDERATION_NAMESPACE } = match.params;
   const onCancel = () => history.goBack();
 
+  const breadcrumbs = [
+    {
+      name: DATA_FEDERATION,
```
Comment (GowthamShanmugam) [inline packages/components/data-resource/create-data-resource.tsx:28]: Same here, need translation

File: packages/utils/details-page/DetailsPage.tsx [hunk 4]
```diff
@@ -82,7 +82,7 @@ const DetailsPage: React.FC<DetailsPageProps> = ({
           }
           actions={actions}
           resource={resource}
-          className="odf-resource-details"
+          className="resource-details"
```
Comment (GowthamShanmugam) [inline packages/utils/details-page/DetailsPage.tsx:85]: same here, the class name needs a prefix. You can use mcgms

File: packages/utils/details-page/details.scss [hunk 5]
```diff
@@ -1,3 +1,3 @@
-.odf-resource-details {
+.resource-details {
```
Comment (GowthamShanmugam) [inline packages/utils/details-page/details.scss:1]: use mcgms as prefix

**Comment by cloudbehl** [comment]
> The bucket creation page with breadcrumbs looks like the below -
> 
> image
> 
> The data source creation page with bread crumbs looks like this -
> 
> image

I think we need padding(specially bottom padding) on the breadcrumbs. Can you check if all the breadcrumbs look the same on other pages?

File: packages/components/bucket-policy/create-bucket-policy.tsx (hunk 2 above)
Comment (debjyoti-pandit) [reply packages/components/bucket-policy/create-bucket-policy.tsx:194]: it was there in UX as OBCs. hopefully, they are aware of it

File: packages/components/bucket-policy/create-bucket-policy.tsx (hunk 1 above)
Comment (debjyoti-pandit) [reply packages/components/bucket-policy/create-bucket-policy.tsx:134]: @SanjalKatiyar we removed the translation for some reason. I am not getting it now. Do you remember it?

File: packages/components/data-resource/create-data-resource.tsx (hunk 3 above)
Comment (debjyoti-pandit) [reply packages/components/data-resource/create-data-resource.tsx:28]: https://github.com/red-hat-storage/mcg-ms-console/pull/43#discussion_r903504981

File: packages/components/bucket-policy/bucket-policy.scss [hunk 6]
```diff
@@ -36,3 +36,7 @@
 .data-resource-modal__check-icon {
   color: var(--pf-global--success-color--100) !important;
 }
+
+.breadcrumbs-header {
```
[the remark this answers is not in the document]
Comment (debjyoti-pandit) [reply packages/components/bucket-policy/bucket-policy.scss:40]: changed to - mcgms-breadcrumbs-header

File: packages/components/data-resource/create-data-resource.tsx [hunk 7]
```diff
@@ -22,22 +23,34 @@ const CreateDataResource: React.FC<CreateDataResourceProps> = ({
...
-          headingLevel="h1"
-          className="co-create-operand__header-text"
-        >
-          {t('Create new data source')}
-        </Title>
-        <p className="help-block">
+      <PageHeading
+        breadcrumbs={breadcrumbs}
+        title={t('Create new data source')}
+        className="breadcrumbs-header"
```
[the remark this answers is not in the document]
Comment (debjyoti-pandit) [reply packages/components/data-resource/create-data-resource.tsx:46]: changed to mcgms-breadcrumbs-header

File: packages/utils/details-page/DetailsPage.tsx (hunk 4 above)
Comment (debjyoti-pandit) [reply packages/utils/details-page/DetailsPage.tsx:85]: changed to mcgms-resource-details

File: packages/utils/details-page/details.scss (hunk 5 above)
Comment (debjyoti-pandit) [reply packages/utils/details-page/details.scss:1]: changed to mcgms-resource-details

**Comment by debjyoti-pandit** [comment]
> > The bucket creation page with breadcrumbs looks like the below -
> > image
> > The data source creation page with bread crumbs looks like this -
> > image
> 
> I think we need padding(especially bottom padding) on the breadcrumbs. Can you check if all the breadcrumbs look the same on other pages?

image

breadcrumbs on the details page also look the same. it has the same padding bottom

**Comment by debjyoti-pandit** [comment]
> 

added padding to both the details page and create page as we were missing it.

image

File: packages/components/bucket-policy/create-bucket-policy.tsx (hunk 2 above)
Comment (GowthamShanmugam) [reply packages/components/bucket-policy/create-bucket-policy.tsx:194]: ok

File: packages/components/data-resource/data-resource-create-form.scss [hunk 8]
```diff
@@ -0,0 +1,3 @@
+.mcgms-breadcrumbs-header {
+  margin-top: var(--pf-global--spacer--md);
+}
```
Comment (SanjalKatiyar) [inline packages/components/data-resource/data-resource-create-form.scss:3]: no need to create new file, we already have one scss file in this directory use that... u can rename that file if u want...

File: packages/utils/heading/page-heading.tsx [hunk 9]
```diff
@@ -83,7 +83,10 @@ const PageHeading: React.FC<PageHeadingProps> = (props) => {
       style={style}
     >
       {showBreadcrumbs && (
-        <Split style={{ alignItems: 'baseline' }}>
+        <Split
+          style={{ alignItems: 'baseline' }}
+          className="mcgms-page-heading-breadcrumbs"
```
Comment (SanjalKatiyar) [inline packages/utils/heading/page-heading.tsx:88]: no need here...
details page also looks okay to me... only it is needed for creation page and that we can do so by passing the class as a prop (which we did above in this PR).

Comment (debjyoti-pandit) [reply packages/utils/heading/page-heading.tsx:88]: I feel with padding it looks better.

image

also, it maintains uniformity, we can keep the padding. I have kept the exact amount of padding we have in various openshift repo - 

image
image

without padding - 

image

**Pull request merged**